components/browser/search/src/main/java/mozilla/components/browser/search/provider/AssetsSearchEngineProvider.kt

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.search.provider
+
+import android.content.Context
+import android.content.res.AssetManager
+import kotlinx.coroutines.experimental.Deferred
+import kotlinx.coroutines.experimental.async
+import mozilla.components.browser.search.SearchEngine
+import mozilla.components.browser.search.SearchEngineParser
+import mozilla.components.browser.search.provider.filter.SearchEngineFilter
+import mozilla.components.browser.search.provider.localization.SearchLocalizationProvider
+import mozilla.components.support.ktx.android.content.res.readJSONObject
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * SearchEngineProvider implementation to load the included search engines from assets.
+ */
+class AssetsSearchEngineProvider(
+    private val localizationProvider: SearchLocalizationProvider,
+    private val filters: List<SearchEngineFilter> = emptyList()
+) : SearchEngineProvider {
+
+    /**
+     * Load search engines from this provider.
+     */
+    override suspend fun loadSearchEngines(context: Context): List<SearchEngine> {
```

Comment:
\[detekt\] INFO: \[style/ModifierOrder\] Modifier order should be: suspend override